diag/diag.proto

```diff
@@ -0,0 +1,238 @@
+    // BERT duration in mins. Must be a positive number.
+    uint32 test_duration_in_mins = 3;  // required
+  }
+  // Unique BERT operation ID specified by the client. Multiple BERTs run on
+  // different ports can have the same BERT operation ID. This ID will be used
+  // later to stop the operation and/or get its results.
+  string bert_operation_id = 1;
+  // All the per-port BERTs that are considered one BERT operation and have the
+  // same BERT operation ID.
+  repeated PerPortRequest per_port_requests = 2;
+}
+
+message StartBERTResponse {
+  // Per-port BERT start responses.
+  message PerPortResponse {
+    // Path to the interface corresponding to the port.
+    gnoi.Path interface = 1;
+    // BERT start status for this port.
+    BertStatus status = 2;
+  }
+  // The same BERT operation ID given by the request.
+  string bert_operation_id = 1;
+  // Captures the results of starting BERT on a per-port basis.
+  repeated PerPortResponse per_port_responses = 2;
+}
+
+message StopBERTRequest {
+  // Per-port BERT stop requests.
+  message PerPortRequest {
+    // Path to the interface corresponding to the port.
+    gnoi.Path interface = 1;
+  }
+  // The same BERT operation ID given when BERT operation was started.
+  string bert_operation_id = 1;
+  // All the per-port BERTs that need to be stopped. Must be part of the BERT
+  // operation specified by the `bert_operation_id` above.
+  repeated PerPortRequest per_port_requests = 2;
+}
+
+message StopBERTResponse {
+  // Per-port BERT stop responses.
+  message PerPortResponse {
+    // Path to the interface corresponding to the port.
+    gnoi.Path interface = 1;
+    // BERT stop status for this port.
+    BertStatus status = 2;
+  }
+  // The same BERT operation ID given by the request.
+  string bert_operation_id = 1;
+  // Captures the results of stopping BERT on a per-port basis.
+  repeated PerPortResponse per_port_responses = 2;
+}
+
+message GetBERTResultRequest {
+  // Per-port BERT get result requests.
+  message PerPortRequest {
+    // Path to the interface corresponding to the port.
+    gnoi.Path interface = 1;
+  }
+  // The same BERT operation ID given when BERT operation was started.
+  string bert_operation_id = 1;
+  // All the per-port BERTs result of which we want to query. Must be part of
+  // the BERT operation specified by the `bert_operation_id` above.
+  repeated PerPortRequest per_port_requests = 2;
+  // If set to true, the results for all the per-port BERTs will be returned.
+  // `bert_operation_id` and `per_port_requests` will be ignored will be
+  // ignored in that case.
+  bool result_from_all_ports = 3;
+}
+
+message GetBERTResultResponse {
+  // Per-port BERT results/status.
+  message PerPortResponse {
+    // Path to the interface corresponding to the port.
+    gnoi.Path interface = 1;
+    // BERT result get status for this port. Only if the status is
+    // BERT_STATUS_OK the rest of the fields below are meaningful.
```

Comment:
Suggest `Only if the status is BERT_STATUS_OK are the rest of the fields meaninful`

Reply:
Done.